indico/modules/core/client/js/impersonation.js

```diff
@@ -17,49 +17,56 @@
 
 /* global ChooseUsersPopup:false */
 
-(function() {
-    'use strict';
+import getImpersonateURL from 'indico-url:auth.admin_impersonate';
 
-    function sendRequest(url, payload) {
-        $.ajax({
-            url: url,
-            method: 'POST',
-            data: payload,
-            complete: IndicoUI.Dialogs.Util.progress(),
-            error: handleAjaxError,
-            success: function() {
-                IndicoUI.Dialogs.Util.progress();
-                location.reload();
-            }
-        });
-    }
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {$T} from 'indico/utils/i18n';
 
-    function impersonateUser(url) {
-        function _userSelected(users) {
-            sendRequest(url, {user_id: users[0].id});
-        }
 
-        var dialog = new ChooseUsersPopup(
-            $T("Select user to impersonate"),
-            true, null, false, true, null, true, true, false, _userSelected, null, false
-        );
+async function sendRequest(data) {
+    const formData = new FormData();
```

Comment:
i guess this is needed because they endpoint doesn't accept a JSON payload? In that case you could switch `RHAdminImpersonate` to webargs. That way both JSON and formdata (which I guess is still needed outside RB) should work fine

Reply:
Indeed. It's probably easier.

Reply:
Done.